src/components/CalendarMonthGrid.jsx

```diff
@@ -111,14 +111,7 @@ export default class CalendarMonthGrid extends React.Component {
     }
 
     if (hasNumberOfMonthsChanged) {
-      if (this.props.numberOfMonths < numberOfMonths) {
-        newMonths = months.slice();
-        for (let i = 0; i < numberOfMonths - this.props.numberOfMonths; i += 1) {
-          newMonths.push(months[months.length - 1].clone().add(i, 'month'));
-        }
-      } else {
-        newMonths = months.slice(0, numberOfMonths + 3);
-      }
+      newMonths = getMonths(initialMonth, numberOfMonths);
```

Comment:
Technically, I think the reason for this was to not create hella new moment objects when you changed the number of months which I think getMonths does, but I suppose that's a fairly rare edgecase.